### PR TITLE
Update Scabbard v3 TimerHandlerFactory and TimerHandler to use the ConsensusRunner

### PR DESCRIPTION
**Pull request by agunde406**
This PR is enables running the ConsensusRunner within the timer handler. As of these changes, setting up a scabbard::v3 circuit tries to come to agreement using TwoPhaseCommit. Currently, the services fail with the following errors.

One side:
```
[2022-05-13 16:23:01.957] T[TimerThread--1] ERROR [splinter::runtime::service::timer::thread] UNIQUE constraint failed: scabbard_v3_commit_history.service_id, scabbard_v3_commit_history.epoch
```

and the other:
```
[4:24](https://bitwiseio.slack.com/archives/C02RHCJ7STG/p1652477046388599)
[2022-05-13 16:23:02.007] T[TimerThread--1] ERROR [splinter::runtime::service::timer::thread] Alarm unexpected in WaitForVoteRequest state
```

Different errors are raised after #1966 goes in

### CODE REVIEW

**Review by peterschwarz — APPROVED** [review]

File: services/scabbard/libscabbard/src/service/v3/timer_handler.rs [hunk 1]
```diff
@@ -12,33 +12,76 @@
...
     ) -> Result<(), InternalError> {
-        info!("handling scabbard timer for: {}", service);
-        Ok(())
+        // Check store if we have an alarm that have expired
+        if let Some(alarm) = self
+            .store
+            .get_alarm(&service, &AlarmType::TwoPhaseCommit)
+            .map_err(|err| InternalError::from_source(Box::new(err)))?
+        {
+            // If the alarm has passed, elapsed will return OK and a duation for how long ago
```
Comment (vaporos) [inline services/scabbard/libscabbard/src/service/v3/timer_handler.rs:60]: Typo - duation

File: services/scabbard/libscabbard/src/service/v3/timer_handler.rs [hunk 2]
```diff
@@ -12,33 +12,76 @@
...
+            // it was. If it is still in the fututr an error is returned with how long until
```
Comment (vaporos) [inline services/scabbard/libscabbard/src/service/v3/timer_handler.rs:61]: Typo fututr

File: services/scabbard/libscabbard/src/service/v3/timer_handler.rs [hunk 3]
```diff
@@ -12,33 +12,76 @@
...
+            if alarm.elapsed().is_ok() {
+                let alarm_event = ConsensusEvent::TwoPhaseCommit(Event::Alarm());
+                let epoch = match self
+                    .store
+                    .get_current_consensus_context(&service)
+                    .map_err(|err| InternalError::from_source(Box::new(err)))?
+                {
+                    Some(ConsensusContext::TwoPhaseCommit(context)) => *context.epoch(),
+                    _ => {
+                        return Err(InternalError::with_message(
```
Comment (vaporos) [inline services/scabbard/libscabbard/src/service/v3/timer_handler.rs:73]: Epoch shouldn't be a factor here, and this runtime error shouldn't be required.

Comment (agunde406) [reply services/scabbard/libscabbard/src/service/v3/timer_handler.rs:73]: This is a larger discussion, as the epoch is required for `store.add_consensus_event`.

File: services/scabbard/libscabbard/src/service/v3/timer_handler.rs (hunk 1 above)
Comment (agunde406) [reply services/scabbard/libscabbard/src/service/v3/timer_handler.rs]: fixed

File: services/scabbard/libscabbard/src/service/v3/timer_handler.rs (hunk 2 above)
Comment (agunde406) [reply services/scabbard/libscabbard/src/service/v3/timer_handler.rs]: fixed

**Review by peterschwarz — DISMISSED** [review]

File: services/scabbard/libscabbard/src/service/v3/timer_handler.rs (hunk 3 above)
Comment (agunde406) [reply services/scabbard/libscabbard/src/service/v3/timer_handler.rs:70]: A story was added to resolve this

Comment (agunde406) [reply services/scabbard/libscabbard/src/service/v3/timer_handler.rs]: that story was complete, so this has been removed

**Review by peterschwarz — APPROVED** [review]

**Pull request merged**